backend/main/src/main/java/sentiment/apps/AppsRequest.java

```diff
@@ -0,0 +1,133 @@
+ */
+public class AppsRequest extends Request {
+  private static final String REGION = System.getenv("DEPLOY_REGION");
+  private static final String TABLE = System.getenv("TABLE_NAME");
+
+  @JsonCreator
+  public AppsRequest() { }
+
+  /**
+   * Process the data for this request and return a response.
+   */
+  public Response process() {
+    // TODO get these from the settings DB
+    final String[] appIds = 
+      new String[] { "com.ford.fordpass*App Store", "com.ford.fordpass*Google Play" };
+
+    List<AppInfo> apps = new ArrayList<AppInfo>();
+
+    AmazonDynamoDB client = AmazonDynamoDBClientBuilder.standard().withRegion(REGION).build();
+
+    for (int i = 0; i < appIds.length; i++) {
+      AppInfo info = getAppInfoFromDb(client, appIds[i]);
+      if (info != null) {
+        // We have reviews for this app
+        apps.add(info);
+      }
+    }
+
+    return new AppsResponse(apps.toArray(new AppInfo[0]));
+  }
+
+  private AppInfo getAppInfoFromDb(AmazonDynamoDB db, String appIdStore) {
+    HashMap<String, AttributeValue> valueMap = new HashMap<String, AttributeValue>();
+    valueMap.put(":id", new AttributeValue().withS(appIdStore));
+
+    // Run two DynamoDB queries - one for max date value and one for min date value
+    // Each query sorts reviews for this app by date and takes the first result
+    QueryRequest minDateQuery = new QueryRequest()
+        .withTableName(TABLE).withIndexName("date")
+        .withKeyConditionExpression("appIdStore = :id")
+        .withExpressionAttributeValues(valueMap)
+        .withLimit(1);
+
+    QueryRequest maxDateQuery = new QueryRequest()
+        .withTableName(TABLE).withIndexName("date")
+        .withKeyConditionExpression("appIdStore = :id")
+        .withExpressionAttributeValues(valueMap)
+        .withLimit(1)
+        .withScanIndexForward(false);
+    
+    QueryResult queryResult = null;
+    String date = null;
+
+    DateTimeFormatter formatter = DateTimeFormatter.ISO_DATE_TIME;
+    LocalDate minDate = null;
+    LocalDate maxDate = null;
+
+    try {
+      queryResult = db.query(minDateQuery);
+      if (queryResult.getCount() == 0) {
+        // No reviews for this app yet
+        return null;
+      }
+
+      date = queryResult.getItems().get(0).get("date").getS();
+      minDate = LocalDate.parse(date, formatter);
+
+      queryResult = db.query(maxDateQuery);
+      if (queryResult.getCount() == 0) {
+        // No reviews for this app yet
+        return null;
+      }
+
+      date = queryResult.getItems().get(0).get("date").getS();
+      maxDate = LocalDate.parse(date, formatter);
+      
+    } catch (Exception exp) {
+      System.err.println("Unable to get date range from DyanmoDB for app " + appIdStore);
+      System.err.println(exp.getMessage());
+    }
+
+    // Run one DynamoDB query to get a list of the version numbers for this app's reviews
+    // The resulting list is then inserted into a HashSet to remove duplicates
```

Comment:
I was going to ask if it was possible to do the "remove duplicates" stuff directly in the query, but just looked it up and saw that's not a thing in DynamoDB. Guess I'm still thinking in terms of SQL!